src/slacko.ml

```diff
@@ -584,10 +584,13 @@ type history_result = [
   | timestamp_error
 ]
 
-let base_url = "https://slack.com/api/"
+let base_url = ref "https://slack.com/api/"
```

Comment:
Thank you for the comprehensive comparison. I have thought a bit about this and agree with your conclusion.

I do not mind as much breaking the API (we are still 0.x so figuring out a decent API is completely okay) nor the code changes in the functor approach but it's true that the functor does not really bring any advantage from the perspective of types, it's just more of a hassle to be able to use it.